java/lang/Float/FloatTest.cpp

```diff
@@ -24,538 +24,1578 @@
-	
-	// Test MAX_VALUE
-	variableTestIntValue = MAX_VALUE;
-	expectedResultIntValue = -2147483648;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_EQUAL(expectedResultIntValue , actualResultIntValue);
-	
-	// Test MIN_NORMAL
-	variableTestIntValue = MIN_NORMAL;
-	expectedResultIntValue = 0;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_EQUAL(expectedResultIntValue , actualResultIntValue);
-	
-	// Test MIN_VALUE
-	variableTestIntValue = MIN_VALUE;
-	expectedResultIntValue = 0;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_EQUAL(expectedResultIntValue , actualResultIntValue);
-	
-	// Test valid case
-	variableTestIntValue = 5.9;
-	expectedResultIntValue = 5;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_EQUAL(expectedResultIntValue , actualResultIntValue);
-	
-	// Test valid case
-	variableTestIntValue = 5.4;
-	expectedResultIntValue = 5;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_EQUAL(expectedResultIntValue , actualResultIntValue);
-	
-	// Test invalid case
-	variableTestIntValue = 6;
-	expectedResultIntValue = 5;
-	actualResultIntValue = variableTestIntValue.intValue();
-	ASSERT_NOT_EQUAL(expectedResultIntValue , actualResultIntValue);
+TEST (JavaLang, FloatIsInfinite2) {
+    Float variableIsInfinite2;
+
+    variableIsInfinite2 = Float::POSITIVE_INFINITY;
+    ASSERT_TRUE(variableIsInfinite2.isInfinite());
+
+    variableIsInfinite2 = Float::NEGATIVE_INFINITY;
+    ASSERT_TRUE(variableIsInfinite2.isInfinite());
+
+    variableIsInfinite2 = 13.02;
+    ASSERT_FALSE(variableIsInfinite2.isInfinite());
 }
 
-TEST(JavaLang, FloatLongValue) {
-	// Create variable to test
-	Float variableTestLongValue;
-	long expectedResultLongValue;
-	long actualResultLongValue;
-	
-	// Test POSITIVE_INFINITY_DOUBLE
-	variableTestLongValue = POSITIVE_INFINITY;
-	expectedResultLongValue = -9223372036854775808;
-	actualResultLongValue = variableTestLongValue.longValue();
-	ASSERT_EQUAL(expectedResultLongValue , actualResultLongValue);
-	
-	// Test NEGATIVE_INFINITY
-	variableTestLongValue = NEGATIVE_INFINITY;
-	expectedResultLongValue = -9223372036854775808;
-	actualResultLongValue = variableTestLongValue.longValue();
-	ASSERT_EQUAL(expectedResultLongValue , actualResultLongValue);
-	
-	// Test NaN_NUMBER
-	variableTestLongValue = NaN_NUMBER;
-	expectedResultLongValue = -9223372036854775808;
-	actualResultLongValue = variableTestLongValue.longValue();
-	ASSERT_EQUAL(expectedResultLongValue , actualResultLongValue);
-	
-	// Test MAX_VALUE
-	variableTestLongValue = MAX_VALUE;
-	expectedResultLongValue = -9223372036854775808;
-	actualResultLongValue = variableTestLongValue.longValue();
-	ASSERT_EQUAL(expectedResultLongValue , actualResultLongValue);
-	
-	// Test MIN_NORMAL
-	variableTestLongValue = MIN_NORMAL;
-	expectedResultLongValue = 0;
-	actualResultLongValue = variableTestLongValue.longValue();
-	ASSERT_EQUAL(expectedResultLongValue , actualResultLo
```

Comment:
Please remove some duplicate test and add test for Comparable

Reply:
Thanks. Fixed.